build/cwl-wrappers/cmo-marianas.ProcessLoopUMIFastq/0.0.0/cmo-marianas.ProcessLoopUMIFastq.original.cwl

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env/cwl-runner
+
+cwlVersion: cwl:v1.0
+
+class: CommandLineTool
+
+baseCommand: [cmo_process_loop_umi_fastq]
+
+arguments: ["-server", "-Xms8g", "-Xmx8g", "-cp"]
+
+doc: Marianas UMI Clipping module
+
+inputs:
+  r1_fastq:
+    type: File
+    inputBinding:
+      prefix: --r1_fastq
+    secondaryFiles:
+    - ${return self.location.replace("_R1_", "_R2_")}
+    - ${return self.location.replace(/(.*)(\/.*$)/, "$1/SampleSheet.csv")}
+
+  umi_length:
+    type: string
+    inputBinding:
+      prefix: --umi_length
+
+  output_project_folder:
+    type: string
+    inputBinding:
+      prefix: --output_project_folder
+
+  outdir:
+    type: ['null', string]
+    doc: Full Path to the output dir.
+    inputBinding:
+      prefix: --outDir
+
+outputs:
+  processed_fastq_1:
+    type: File
+    outputBinding:
+      glob: ${ return "**/" + inputs.r1_fastq.basename }
+
+  processed_fastq_2:
+    type: File
+    outputBinding:
+      glob: ${ return "**/" + inputs.r1_fastq.basename.replace("_R1_", "_R2_") }
+
+  composite_umi_frequencies:
+    type: File
+    outputBinding:
+      glob: ${ return "**/composite-umi-frequencies.txt" }
+
+  info:
+    type: File
+    outputBinding:
+      glob: ${ return "**/info.txt" }
+
+  sample_sheet:
+    type: File
+    outputBinding:
+      glob: ${ return "**/SampleSheet.csv" }
+
+  umi_frequencies:
+    type: File
+    outputBinding:
+      glob: ${ return "**/umi-frequencies.txt" }
```

Comment:
I would like to specify the exact output folder instead of `**`, however when I try the foldername that is created by the Marianas tool, I get an error:

```
Did not find output file with glob pattern: '[u'Sample_MSK-L-007-cf_IGO_05500_DY_22--small/*.txt']'
```

I'm not sure how to figure out what the exact directory structure is for the output files after the completed job, but maybe I have to use the temporary folder names like `tmpAbYAuQ/stg68388e60-f306-4a7d-81b1-935c9608356e/*.txt`?

Reply:
can you show me the actual cwl that is generating that error. it looks like an array is being passed where it shouldn't be.

Reply:
```
   doc: Marianas UMI Clipping module

   inputs:
     r1_fastq:
       type: File
       inputBinding:
         prefix: --r1_fastq
       secondaryFiles:
       - ${return self.location.replace("_R1_", "_R2_")}
       - ${return self.location.replace(/(.*)(\/.*$)/, "$1/SampleSheet.csv")}

     umi_length:
       type: string
       inputBinding:
         prefix: --umi_length

     output_project_folder:
       type: string
       inputBinding:
         prefix: --output_project_folder

     outdir:
       type: ['null', string]
       doc: Full Path to the output dir.
       inputBinding:
         prefix: --outDir

   outputs:
     processed_fastq_1:
       type: File
       outputBinding:
         glob: ${ return "**/" + inputs.r1_fastq.basename }

     processed_fastq_2:
       type: File
       outputBinding:
         glob: ${ return "**/" + inputs.r1_fastq.basename.replace("_R1_", "_R2_") }

     composite_umi_frequencies:
       type: File
       outputBinding:
         glob: ${ return "**/composite-umi-frequencies.txt" }

     info:
       type: File
       outputBinding:
         glob: ${ return "**/info.txt" }

     sample_sheet:
       type: File
       outputBinding:
         glob: ${ return "**/SampleSheet.csv" }

     umi_frequencies:
       type: File
       outputBinding:
         glob: ${ return "Sample_MSK-L-007-cf_IGO_05500_DY_22--small/umi-frequencies.txt" }
```


```
INFO:cwltool:[job cmo-marianas.ProcessLoopUMIFastq.cwl] /home/johnsoni/umi-trimming-module/outputs/tmp4FmCZr$ cmo_process_loop_umi_fastq \
    -server \
    -Xms8g \
    -Xmx8g \
    -cp \
    --output_project_folder \
    . \
    --r1_fastq \
    /home/johnsoni/umi-trimming-module/outputs/tmpqEtgT7/stgec1b9fcc-d327-47b3-96c3-7a7c52894b3e/MSK-L-007-cf_IGO_05500_DY_22_S23_R1_001.fastq.gz--small.fastq.gz \
    --umi_length \
    3
2017-11-21 14:42:58,197 call_cmd INFO EXECUTING: sing-java.sh -server -Xms8g -Xmx8g -cp sing.sh marianas 0.0.0 org.mskcc.marianas.umi.duplex.fastqprocessing.ProcessLoopUMIFastq /home/johnsoni/umi-trimming-module/outputs/tmpqEtgT7/stgec1b9fcc-d327-47b3-96c3-7a7c52894b3e/MSK-L-007-cf_IGO_05500_DY_22_S23_R1_001.fastq.gz--small.fastq.gz 3 .
[job cmo-marianas.ProcessLoopUMIFastq.cwl] Job error:
../../../ifs/work/bergerm1/Innovation/sandbox/ian/roslin-pipelines/variant/1.3.1/bin/cwl/cmo-marianas.ProcessLoopUMIFastq/0.0.0/cmo-marianas.ProcessLoopUMIFastq.cwl:105:3: Error collecting output for parameter 'umi_frequencies':
../../../ifs/work/bergerm1/Innovation/sandbox/ian/roslin-pipelines/variant/1.3.1/bin/cwl/cmo-marianas.ProcessLoopUMIFastq/0.0.0/cmo-marianas.ProcessLoopUMIFastq.cwl:108:7:   Did not find output file with glob pattern: '[u'Sample_MSK-L-007-cf_IGO_05500_DY_22--small/umi-frequencies.txt']'
ERROR:cwltool:[job cmo-marianas.ProcessLoopUMIFastq.cwl] Job error:
../../../ifs/work/bergerm1/Innovation/sandbox/ian/roslin-pipelines/variant/1.3.1/bin/cwl/cmo-marianas.ProcessLoopUMIFastq/0.0.0/cmo-marianas.ProcessLoopUMIFastq.cwl:105:3: Error collecting output for parameter 'umi_frequencies':
../../../ifs/work/bergerm1/Innovation/sandbox/ian/roslin-pipelines/variant/1.3.1/bin/cwl/cmo-marianas.ProcessLoopUMIFastq/0.0.0/cmo-marianas.ProcessLoopUMIFastq.cwl:108:7:   Did not find output file with glob pattern: '[u'Sample_MSK-L-007-cf_IGO_05500_DY_22--small/umi-frequencies.txt']'
[job cmo-marianas.ProcessLoopUMIFastq.cwl] completed permanentFail
WARNING:cwltool:[job cmo-marianas.ProcessLoopUMIFastq.cwl] completed permanentFail
Traceback (most recent call last):
  File "/opt/common/CentOS_6-dev/python/python-2.7.10/lib/python2.7/site-packages/toil-3.8.0a1-py2.7.egg/toil/worker.py", line 340, in main
    job._runner(jobGraph=jobGraph, jobStore=jobStore, fileStore=fileStore)
  File "/opt/common/CentOS_6-dev/python/python-2.7.10/lib/python2.7/site-packages/toil-3.8.0a1-py2.7.egg/toil/job.py", line 1289, in _runner
    returnValues = self._run(jobGraph, fileStore)
  File "/opt/common/CentOS_6-dev/python/python-2.7.10/lib/python2.7/site-packages/toil-3.8.0a1-py2.7.egg/toil/job.py", line 1234, in _run
    return self.run(fileStore)
  File "/opt/common/CentOS_6-dev/python/python-2.7.10/lib/python2.7/site-packages/toil-3.8.0a1-py2.7.egg/toil/cwl/cwltoil.py", line 311, in run
    raise cwltool.errors.WorkflowException(status)
WorkflowException: permanentFail
ERROR:toil.worker:Exiting the worker because of a failed job on host u09.cbio.private
WARNING:toil.jobGraph:Due to failure we are reducing the remaining retry count of job 'cmo_process_loop_umi_fastq' cmo_process_loop_umi_fastq k/a/jobk5r48q with ID k/a/jobk5r48q to 0
WARNING:toil.jobGraph:We have increased the default memory of the failed job 'cmo_process_loop_umi_fastq' cmo_process_loop_umi_fastq k/a/jobk5r48q to 21474836480 bytes
```

Reply:
is the file supplied to the command line named like you expect? 

MSK-L-007-cf_IGO_05500_DY_22_S23_R1_001.fastq.gz--small.fastq.gz

it seems like that might not be what you actually want to pass marianas